### ISSUE TITLE
Enhancement request: Option to be lenient on the first indent in procedure or function 

### ISSUE THREAD

**Issue by cverbiest** [Riverside-Software/sonar-openedge#1084]
We have a large number of false positives on code that was maintained using the AppBuilder .
Within an appbuilder section editor we did not start with an indent because in that UI this seemed unnecessary.

I do not want to disable the eu.rssw.antlr.proparse.checks.IndentationCheck rule as it is valuable because it can detect hard to spot errors.

Would it be possible to add an option to not flag the first statement within a procedure or function ?

Example
```
&ANALYZE-SUSPEND _UIB-CODE-BLOCK _PROCEDURE FindParent W-Win
PROCEDURE FindParent :
/*------------------------------------------------------------------------------
  Purpose:
  Parameters:  <none>
  Notes:
------------------------------------------------------------------------------*/
if condition
then do:
    statement.
end.

end procedure.

/* _UIB-CODE-BLOCK-END */
&ANALYZE-RESUME

```

**Comment by gquerret** [comment]
An option to always skip the first statement of a procedure or function seems a bit too much to my mind. But restricting that to UIB code seems to specific to your codebase.  Not sure for now about what would be the best way to handle that.

**Comment by cverbiest** [comment]
I wouldn't mind if it applied to all first statements of functions, methods, procedures.

**Comment by gquerret** [comment]
Even for methods ? I'd assume that the syntax is only for code opened in the AppBuilder, so indentation should be correct in methods

**Comment by gquerret** [comment]
Available in 2.24
